src/test/iv_client.c

```diff
@@ -406,6 +406,39 @@ progress_function(void *data)
 	return NULL;
 }
 
+static struct crt_proto_rpc_format my_proto_rpc_fmt_iv[] = {
+	{
+		.prf_flags	= 0,
+		.prf_req_fmt	= &CQF_RPC_TEST_FETCH_IV,
+		.prf_hdlr	= NULL,
+		.prf_co_ops	= NULL,
+	}, {
+		.prf_flags	= 0,
+		.prf_req_fmt	= &CQF_RPC_TEST_UPDATE_IV, 
+		.prf_hdlr	= NULL,
+		.prf_co_ops	= NULL,
+	}, {
+		.prf_flags	= 0,
+		.prf_req_fmt	= &CQF_RPC_TEST_INVALIDATE_IV, 
+		.prf_hdlr	= NULL,
+		.prf_co_ops	= NULL,
+	}, {
+		.prf_flags	= 0,
```

Comment:
this table should be shared between client and server. without that you can easily get out of sync between client and server, which happened now as well with 'set_ivns' call not being declared for client.

Reply:
It depends on what you're trying to achieve, the protocol code can either allow version interoperability or it can simply enforce non-compatible versions from being run.  If you don't need the client to work with different servers then the table can be shared between client and server, and all we need to do is change the version number when the RPCs change, if the client is intended to ever support multiple versions then they need to be seperate.

For the tests I suspect as they're intended to be installed as part of the same RPM simply sharing the table is correct, and only updating the version number on every release rather than evert change is sufficient.

For other code where the client/server are in different RPMs, or will be built into different applications installed on different nodes this needs to be done properly where the client supports multiple versions and queries the server using the proto_query() function.  The self_test program and protocol is an example of where we'd need to support this usage for example.

As this is a test file @frostedcmos is correct, but this is not true for the general case.

Reply:
The client and server have different handlers, so sharing the tables will not work. It is the opcodes that have to be common and adjusted to similar sized tables.

Reply:
That's another point.  In IOF we used the same structs, but the server then passed in a array of handlers which was used to modify the struct before registration, but that didn't seem a particularly good way.